src/main/java/org/spongepowered/api/command/CommandCallable.java

```diff
@@ -60,10 +64,13 @@
      *
      * @param source The command source
      * @param arguments The arguments entered up to this point
+     * @param location The {@link Location} being looked at by the user,
+     * if available
+     *
      * @return A list of suggestions
      * @throws CommandException Thrown if there was a parsing error
      */
-    List<String> getSuggestions(CommandSource source, String arguments) throws CommandException;
+    List<String> getSuggestions(CommandSource source, String arguments, @Nullable Location<World> location) throws CommandException;
```

Comment:
@zml2008: Do you think it should be removed entirely (e.g. from `TabCompleteEvent`), or just from the command system?